apps/worker/src/app/workflow/usecases/handle-last-failed-job/handle-last-failed-job.usecase.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@nestjs/common';
+import { JobRepository } from '@novu/dal';
+import { ExecutionDetailsSourceEnum, ExecutionDetailsStatusEnum } from '@novu/shared';
+import * as Sentry from '@sentry/node';
+import {
+  CreateExecutionDetails,
+  CreateExecutionDetailsCommand,
+  DetailEnum,
+  InstrumentUsecase,
+} from '@novu/application-generic';
+
+import { HandleLastFailedJobCommand } from './handle-last-failed-job.command';
+
+import { QueueNextJob, QueueNextJobCommand } from '../queue-next-job';
+import { RunJob, RunJobCommand } from '../run-job';
+import { SendMessage, SendMessageCommand } from '../send-message';
+
+@Injectable()
+export class HandleLastFailedJob {
+  constructor(
+    private createExecutionDetails: CreateExecutionDetails,
+    private queueNextJob: QueueNextJob,
+    private runJob: RunJob,
+    private jobRepository: JobRepository
+  ) {}
+
+  /**
+   * TODO: IMPORTANT
+   * We are going to assume that is not very likely that a job has to be retried the maximum default attempts.
+   * That means that we could do a database call here to retrieve the job information for the ExecutionDetails
+   * as it will be only happening for a small percentage of failed jobs.
+   * This consideration is sensible as we are afraid of having to execute many database calls every time the
+   * WorkflowQueueService pulls a job to run or queue again.
+   */
```

Comment:
I don't know if we need this TODO here as it's not accurate because we do run many database calls when the job is executed. Or maybe you know something and can share more context?

Reply:
Maybe I should rephrase it in the way that this is trying to avoid an extra overload of database calls by not requesting the Job entity every time a job is pulled in by the worker and it fails and just when has to back off because is when we set the execution details log about it and we try to queue the next one if existing.
I wanted to have a balance and not to overload MongoDB to alleviate the Redis instance, so for this extra Mongo DB call that we would need to operate this part of the flow, might not be even executed because the case scenario of jobs reaching the maximum attempts and that have to back off should be reduced compared to other outcomes of the flow.